samples/snippets/src/main/java/com/example/bigtable/WriteBatch.java

```diff
@@ -32,34 +32,32 @@ public static void writeBatch(String projectId, String instanceId, String tableI
     // String tableId = "mobile-time-series";
 
     try (BigtableDataClient dataClient = BigtableDataClient.create(projectId, instanceId)) {
-      long timestamp = System.currentTimeMillis() * 1000;
+      try (Batcher<RowMutationEntry, Void> batcher = dataClient.newBulkMutationBatcher(tableId)) {
+        long timestamp = System.currentTimeMillis() * 1000;
 
-      BulkMutation bulkMutation =
-          BulkMutation.create(tableId)
-              .add(
-                  "tablet#a0b81f74#20190501",
-                  Mutation.create()
-                      .setCell(
-                          COLUMN_FAMILY_NAME,
-                          ByteString.copyFrom("connected_wifi".getBytes()),
-                          timestamp,
-                          1)
-                      .setCell(COLUMN_FAMILY_NAME, "os_build", timestamp, "12155.0.0-rc1"))
-              .add(
-                  "tablet#a0b81f74#20190502",
-                  Mutation.create()
-                      .setCell(
-                          COLUMN_FAMILY_NAME,
-                          ByteString.copyFrom("connected_wifi".getBytes()),
-                          timestamp,
-                          1)
-                      .setCell(COLUMN_FAMILY_NAME, "os_build", timestamp, "12155.0.0-rc6"));
+        batcher.add(
+                RowMutationEntry.create("tablet#a0b81f74#20190501")
+                        .setCell(COLUMN_FAMILY_NAME,
+                                ByteString.copyFrom("connected_wifi".getBytes()),
+                                timestamp,
+                                1)
+                        .setCell(COLUMN_FAMILY_NAME, "os_build", timestamp, "12155.0.0-rc1"));
+        batcher.add(
+                RowMutationEntry.create("tablet#a0b81f74#20190502")
+                        .setCell(COLUMN_FAMILY_NAME,
+                                ByteString.copyFrom("connected_wifi".getBytes()),
+                                timestamp,
+                                1)
+                        .setCell(COLUMN_FAMILY_NAME, "os_build", timestamp, "12155.0.0-rc6"));
 
-      dataClient.bulkMutateRows(bulkMutation);
+        // Blocks until mutations are applied on all submitted row entries.
+        // flush will be called automatically when a batch is full.
+        batcher.flush();
+      } // Before batcher is closed, all remaining (if any) mutations are applied.
 
       System.out.print("Successfully wrote 2 rows");
     } catch (Exception e) {
```

Comment:
You should catch a BatchingException here

Reply:
Also iterate over the api futures and print out the row key and the error. Something like:

```
catch (BatchingException e) {
  System.out.println("At least one entry failed to apply. Summary of the errors: " + e.getMessage());
  // Print individual errors:
  for(ApiFuturee f : futures) {
    try {
      f.get();
    } catch(ExecutionException e) {
       System.out.println("Entry failure: " + e.getCause());
    }
  }
}
```